Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -429,14 +429,26 @@ public final class OAuth2Client {
                 return
             }
             
-            // Perform idToken/accessToken validation
-            do {
-                try response.result.validate(using: self, with: request as? IDTokenValidatorContext)
-            } catch {
-                completion(.failure(.validation(error: error)))
-                return
+            group.enter()
+            self.openIdConfiguration { result in
```

Comment:
This is still serializing the requests, since you're kicking off the request for openIdConfiguration within the `request.send` completion block.  Can you use the fetch for the keySet up on line 415 as a guide?

Reply:
Thanks for pointing me in that direction. I have updated this.